src/pages/machine/TuringEdge.tsx

```diff
@@ -17,24 +17,29 @@ const EditableDiv = ({
   const contentEditableRef = useRef(null);
 
   useEffect(() => {
-    // Set the content of the div when edgeValue changes
     if (contentEditableRef.current) {
       // @ts-ignore
-      contentEditableRef.current.textContent = edgeValue;
+      const currentText = contentEditableRef.current.textContent;
+      if (currentText !== edgeValue) {
+        // @ts-ignore
+        contentEditableRef.current.textContent = edgeValue;
+      }
     }
   }, [edgeValue]);
 
   const handleInput = () => {
     if (contentEditableRef.current) {
-      // Update value in parent component when content changes
-      // @ts-ignore
-      onChangeHandler(contentEditableRef.current.textContent);
+      if (contentEditableRef.current) {
+        // @ts-ignore
+        const text = contentEditableRef.current.textContent || '';
+        onChangeHandler(text);
+      }
     }
   };
 
   return (
     <div
-      contentEditable="plaintext-only"
+      contentEditable="true"
```

Comment:
Due to the incompatibility mentioned here:
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable

I'm not the biggest fan of allowing non-plain-text input, but Firefox support is important so happy for this to go through 😃